javascript/src/proxies.ts

```diff
@@ -948,3 +964,8 @@ function assertString(value: Text | string): asserts value is string {
     throw new Error("value was not a string")
   }
 }
+
+function printPath(path: Prop[]): string {
+  // print the path as a json pointer
+  return "/" + path.map(encodeURIComponent).join("/")
```

Comment:
Feels a little more ergonomic to use dot notation IMO, wdyt?

```suggestion
  return "." + path.map(encodeURIComponent).join(".")
```

Reply:
Yeah that's probably a good shout, although at that point I should remove the comment about it being a JSON pointer (I mean, it's not really anyway because of the `encodeURIComponent` bit, maybe it's fine for this to be a bit ambiguous if there are key names with dots in them, it's just an error message)

Reply:
Actually, I've changed my mind. I think using JSON pointer is probably a bit better than making up a scheme which will in the end be quite similar to JSON pointer. (I was thinking about array indices, which look bad using a `.` separator and then thinking that I can't just use `[<index>]` because I'm sure there's a fair number of keys out there with `[` in them and then I ran out of energy and figured it's probably best to use something that is at least vaguely standardized.